common/date.go

```diff
@@ -0,0 +1,43 @@
+package common
+
+import (
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// ParseDate parses a date string of the format YYYY-MM-DD
+func ParseDate(date string) (t time.Time, err error) {
+	layout := "2006-01-02" //this represents YYYY-MM-DD
+	return time.Parse(layout, date)
+}
+
+// ParseDateRange parses a date range string of the format start:end
+//   where the start and end date are of the format YYYY-MM-DD.
+//   The parsed dates are *time.Time and will return nil pointers for
```

Comment:
Why do we return pointers? Is it so we can use nil? Is the zero time and IsZero not sufficient?